timemachine/cpp/src/kernels/k_neighborlist.cuh

```diff
@@ -246,19 +264,41 @@ void __global__ k_find_blocks_with_ixns(
         RealType pos_j_y = atom_j_idx < N ? coords[atom_j_idx*3 + 1] : 0;
         RealType pos_j_z = atom_j_idx < N ? coords[atom_j_idx*3 + 2] : 0;
 
+        if (single_periodic_box) {
+	    // Recenter using **row** box center
+            pos_j_x -= bx*nearbyint((pos_j_x-row_bb_ctr_x)*inv_bx);
+            pos_j_y -= by*nearbyint((pos_j_y-row_bb_ctr_y)*inv_by);
+            pos_j_z -= bz*nearbyint((pos_j_z-row_bb_ctr_z)*inv_bz);
+
+            non_periodic_dist_j = static_cast<RealType>(0.5) * (pos_j_x*pos_j_x + pos_j_y*pos_j_y + pos_j_z*pos_j_z);
+        }
+
+
         unsigned includeAtomFlags = 0;
         while(atomFlags) {
             const int row_atom = __ffs(atomFlags)-1;
             atomFlags &= atomFlags-1;
-
-            RealType atom_atom_dx = __shfl_sync(FULL_MASK, pos_i_x, row_atom) - pos_j_x;
-            RealType atom_atom_dy = __shfl_sync(FULL_MASK, pos_i_y, row_atom) - pos_j_y;
-            RealType atom_atom_dz = __shfl_sync(FULL_MASK, pos_i_z, row_atom) - pos_j_z;
-
-            atom_atom_dx -= bx*nearbyint(atom_atom_dx*inv_bx);
-            atom_atom_dy -= by*nearbyint(atom_atom_dy*inv_by);
-            atom_atom_dz -= bz*nearbyint(atom_atom_dz*inv_bz);
-            interacts |= atom_atom_dx*atom_atom_dx + atom_atom_dy*atom_atom_dy + atom_atom_dz*atom_atom_dz < cutoff_squared;
+            RealType row_i_x = __shfl_sync(FULL_MASK, pos_i_x, row_atom);
+            RealType row_i_y = __shfl_sync(FULL_MASK, pos_i_y, row_atom);
+            RealType row_i_z = __shfl_sync(FULL_MASK, pos_i_z, row_atom);
+            if (!single_periodic_box) {
+                RealType atom_atom_dx = row_i_x - pos_j_x;
+                RealType atom_atom_dy = row_i_y - pos_j_y;
+                RealType atom_atom_dz = row_i_z - pos_j_z;
+
+                atom_atom_dx -= bx*nearbyint(atom_atom_dx*inv_bx);
+                atom_atom_dy -= by*nearbyint(atom_atom_dy*inv_by);
+                atom_atom_dz -= bz*nearbyint(atom_atom_dz*inv_bz);
+
+                interacts |= atom_atom_dx*atom_atom_dx + atom_atom_dy*atom_atom_dy + atom_atom_dz*atom_atom_dz < cutoff_squared;
+            } else {
+		// All threads in warp need single_periodic_box to be true for this not to hang
+		RealType corrected_i = __shfl_sync(FULL_MASK, non_periodic_dist_i, row_atom);
```

Comment:
Of all of the interactions found in tests/test_nblist.py (91866 total) about 46% of them (42230) go through this path